cmake/developer_package/compile_flags/sanitizer.cmake

```diff
@@ -5,20 +5,36 @@
 include(CheckCXXCompilerFlag)
 
 if (ENABLE_SANITIZER)
-    set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} -fsanitize=address")
-    check_cxx_compiler_flag("-fsanitize-recover=address" SANITIZE_RECOVER_ADDRESS_SUPPORTED)
-    if (SANITIZE_RECOVER_ADDRESS_SUPPORTED)
-        set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} -fsanitize-recover=address")
+    check_cxx_compiler_flag("/fsanitize=address" SANITIZE_ADDRESS_SUPPORTED)
+    if (SANITIZE_ADDRESS_SUPPORTED)
+        if (WIN32)
+            set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} /fsanitize=address")
+            check_cxx_compiler_flag("/fsanitize-recover=address" SANITIZE_RECOVER_ADDRESS_SUPPORTED)
```

Comment:
Is VisualStudio support sanitize-recover? If not, then better to remove it until support be implemented.

https://developercommunity.visualstudio.com/t/add-fsanitize-recoveraddress-support-to-asan/1459414

Reply:
Removed redundant check